packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -259,1548 +169,327 @@ class UpdatePackagesCommand extends FlutterCommand {
-    // magic comment flagging them as auto-generated transitive dependencies
-    // that we added in a previous run.
-    return inputData.whereType<PubspecDependency>().where(
-      (PubspecDependency data) =>
-          data.kind != DependencyKind.overridden && !data.isTransitive && !data.isDevDependency,
-    );
-  }
-
-  /// This returns all regular dependencies and all dev dependencies.
-  Iterable<PubspecDependency> get allExplicitDependencies {
-    return inputData.whereType<PubspecDependency>().where(
-      (PubspecDependency data) => data.kind != DependencyKind.overridden && !data.isTransitive,
-    );
   }
 
-  /// This returns all dependencies.
-  Iterable<PubspecDependency> get allDependencies {
-    return inputData.whereType<PubspecDependency>();
-  }
-
-  /// Take a dependency graph with explicit version numbers, and apply them to
-  /// the pubspec.yaml, ignoring any that we know are special dependencies (those
-  /// that depend on the Flutter or Dart SDK directly and are thus automatically
-  /// pinned).
-  void apply(PubDependencyTree versions, Set<String> specialDependencies) {
-    final List<String> output = <String>[]; // the string data to output to the file, line by line
-    final Set<String> directDependencies =
-        <String>{}; // packages this pubspec directly depends on (i.e. not transitive)
-    final Set<String> devDependencies = <String>{};
-    Section section = Section.other; // the section we're currently handling
-
-    // the line number where we're going to insert the transitive dependencies.
-    int? endOfDirectDependencies;
-    // The line number where we're going to insert the transitive dev dependencies.
-    int? endOfDevDependencies;
-    // Walk the pre-parsed input file, outputting it unmodified except for
-    // updating version numbers, removing the old transitive dependencies lines,
-    // and adding our new transitive dependencies lines. We also do a little
-    // cleanup, removing trailing spaces, removing double-blank lines, leading
-    // blank lines, and trailing blank lines, and ensuring the file ends with a
-    // newline. This cleanup lets us be a little more aggressive while building
-    // the output.
-    for (final PubspecLine data in inputData) {
-      if (data is PubspecHeader) {
-        // This line was a header of some sort.
-        //
-        // If we're leaving one of the sections in which we can list transitive
-        // dependencies, then remember this as the current last known valid
-        // place to insert our transitive dependencies.
-        if (section == Section.dependencies) {
-          endOfDirectDependencies = output.length;
+  Map<String, Map<String, String>> _fetchDeps(YamlEditor yamlEditor) {
+    final Map<String, Map<String, String>> dependencies = <String, Map<String, String>>{};
+    for (final String depType in <String>['dependencies', 'dev_dependencies']) {
```

Comment:
Do we have constants for `dependencies` and `dev_dependencies`? It might be worth considering adding them.